satpy/readers/fci_l1c_nc.py

```diff
@@ -700,6 +709,31 @@ def calibrate_rad_to_refl(self, radiance, key):
         res = 100 * radiance * np.float32(np.pi) * np.float32(sun_earth_distance) ** np.float32(2) / cesi
         return res
 
+    def _collect_listed_variables(self, file_handle, listed_variables, filetype_info):
+        listed_variables = self._recycle_shared_info(listed_variables, filetype_info)
+        super()._collect_listed_variables(file_handle, listed_variables, filetype_info)
+        self._store_shared_info(filetype_info)
+
+    def _recycle_shared_info(self, listed_variables, filetype_info):
+        if "shared_info" in filetype_info:
+            shared_info = filetype_info["shared_info"]
+            for key in shared_info:
+                self.file_content[key] = shared_info[key]
+                try:
+                    listed_variables.remove(key)
+                except ValueError:
+                    pass
+        return listed_variables
+
+    def _store_shared_info(self, filetype_info):
+        if "shared_info" not in filetype_info:
+            shared_info = {}
+            for key in self.file_content:
+                if any(key.endswith(k) for k in NONSHAREABLE_VARIABLE_ENDINGS):
+                    continue
+                shared_info[key] = self.file_content[key]
+            filetype_info["shared_info"] = shared_info
```

Comment:
Is `filetype_info` really an appropriate place to put this cache?  I fear this might be confusing.

Reply:
We talked about this with @ameraner at PCW and this was already used in somewhere. LI L2, perhaps? It was also the path of least resistance as it was already in place and the same dict is passed between different filehandlers by the YAML reader.

If there are other backwards compatible ways to pass the info between the file handlers then let me know.